com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -816,6 +817,28 @@ public virtual void Heuristic(float[] actionsOut)
         /// </summary>
         internal void InitializeSensors()
         {
+            // Iterate over Observables
+            var fields = this.GetType().GetFields(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
TODO can we filter out the base Agent fields and properties? Adding BindingFlags.DeclaredOnly might filter out too much.

Reply:
Done (optionally)